yarn-project/simulator/src/avm/opcodes/conversion.test.ts

```diff
@@ -0,0 +1,82 @@
+import { type AvmContext } from '../avm_context.js';
+import { type Uint8, Uint32 } from '../avm_memory_types.js';
+import { initContext, randomMemoryFields } from '../fixtures/index.js';
+import { Addressing, AddressingMode } from './addressing_mode.js';
+import { ToRadixLE } from './conversion.js';
+
+describe('Conversion Opcodes', () => {
+  let context: AvmContext;
+
+  beforeEach(async () => {
+    context = initContext();
+  });
+
+  describe('To Radix LE', () => {
+    it('Should (de)serialize correctly', () => {
+      const buf = Buffer.alloc(18);
+      buf.writeUInt8(ToRadixLE.opcode, 0); // Opcode
+      buf.writeUInt8(1, 1); // Indirect
+      buf.writeUint32BE(0x123, 2); // src memory address
+      buf.writeUint32BE(0x345, 6); // dst memory address
+      buf.writeUint32BE(2, 10); // radix (immediate)
+      buf.writeUint32BE(256, 14); // number of limbs (Immediate)
+      const inst = new ToRadixLE(
+        /*indirect=*/ 1,
+        /*srcOffset=*/ 0x123,
+        /*dstOffset=*/ 0x345,
+        /*radix=*/ 2,
+        /*numLimbs=*/ 256,
+      );
+
+      expect(ToRadixLE.deserialize(buf)).toEqual(inst);
+      expect(inst.serialize()).toEqual(buf);
+    });
+
+    it('Should decompose correctly - direct', async () => {
+      const [arg] = randomMemoryFields(1);
+      const indirect = 0;
+      const srcOffset = 0;
+      const radix = 2; // Bit decomposition
+      const numLimbs = 10; // only the first 10 bits
+      const dstOffset = 20;
+      context.machineState.memory.set(srcOffset, arg);
+
+      await new ToRadixLE(indirect, srcOffset, dstOffset, radix, numLimbs).execute(context);
+
+      const resultBuffer: Buffer = Buffer.concat(
+        context.machineState.memory.getSliceAs<Uint8>(dstOffset, numLimbs).map(byte => byte.toBuffer()),
+      );
+      const expectedBuffer: Buffer = Buffer.from(
+        Uint8Array.from({ length: numLimbs }).map((_, i) => Number((arg.toBigInt() >> BigInt(i)) & 1n)),
```

Comment:
I'd prefer it if you use a non-random field and have a fixed array here. You may not even need the conversion to buffer.

E.g., you can pass `new Field(0b1011101010100n)` as arg, and then compare with [1,0,...].map(i => new Field(i)], or similar.